code/game/machinery/autolathe.dm

```diff
@@ -215,53 +269,63 @@
 			return
 		materials_needed[material] = amount_needed
 
+	//checks for available materials
 	var/material_cost_coefficient = ispath(design.build_path, /obj/item/stack) ? 1 : creation_efficiency
 	if(!materials.has_materials(materials_needed, material_cost_coefficient, build_count))
 		say("Not enough materials to begin production.")
 		return
 
-	//use power
-	var/total_charge = 0
+	//compute power & time to print 1 item
+	var/charge_per_item = 0
 	for(var/material in design.materials)
-		total_charge += round(design.materials[material] * material_cost_coefficient * build_count)
-	var/charge_per_item = total_charge / build_count
-
-	var/total_time = (design.construction_time * design.lathe_time_factor * build_count) ** 0.8
-	var/time_per_item = total_time / build_count
-	start_making(design, build_count, time_per_item, material_cost_coefficient, charge_per_item)
-	return TRUE
-
-/// Begins the act of making the given design the given number of items
-/// Does not check or use materials/power/etc
-/obj/machinery/autolathe/proc/start_making(datum/design/design, build_count, build_time_per_item, material_cost_coefficient, charge_per_item)
```

Comment:
I don't think that moving contents from a separate proc into one that's already over a hundred lines long is a code improvement. That just makes that long proc, even longer, and if we somehow wanted to have another way to start making an item from the autolathe without using the UI, we no longer have that option anymore.

This feels like the inverse of a code improvement, imo.

Reply:
It's a code improvement in the sense we have don't have a chain of proc calls saving us a little overhead 

We should use procs only if that code is being used in atleast more than 1 place which wasn't the case here 

> and if we somehow wanted to have another way to start making an item from the autolathe without using the UI

There is no such use case here so I don't see a need for that, We don't plan code for a future project